src/lib.rs

```diff
@@ -8,18 +8,10 @@ struct BuildDir {
 
 impl BuildDir {
     fn new() -> Self {
-        let mut random_data = [0u8; 16];
-        let mut file = fs::File::open("/dev/urandom").expect("failed to open /dev/urandom");
-        file.read_exact(&mut random_data)
-            .expect("failed to read /dev/urandom");
-
-        let mut hex_str = String::new();
-        for digit in random_data.iter() {
-            hex_str = hex_str + &format!("{:x}", digit)
-        }
+        let now = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH).expect("Cannot compute duration since UNIX epoch");
 
         let mut dir = env::temp_dir();
-        dir.push(format!("build-script-{}", hex_str));
+        dir.push(format!("build-script-{}", now.as_secs()));
```

Comment:
I don't think epoch seconds is quite unique enough here; we use this in quite a few crates internally, and I could easily imagine two of them building at once.